src/main/groovy/com/bmuschko/gradle/docker/DockerRemoteApiPlugin.groovy

```diff
@@ -43,7 +43,9 @@ class DockerRemoteApiPlugin implements Plugin<Project> {
         // if no repositories were defined fallback to buildscript
         // repositories to resolve dependencies as a last resort
         if (project.repositories.size() == 0) {
```

Comment:
I still don't quite understand the rationale why the repositories are added, but should the check on the number of repositories then be done also in `afterEvaluate`? Otherwise it will apply even if the user added project repositories. The only possibility I can see right now for project repositories to have been added at that point would be if another plugin, that has been applied before the Docker plugin, added project repositories already.